src/module/item/base/document.ts

```diff
@@ -791,6 +774,48 @@ class ItemPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends Item
         return super._preUpdate(changed, options, user);
     }
 
+    /** Store certain data to be checked in _onUpdateOperation */
+    static override async _preUpdateOperation(
+        documents: foundry.abstract.Document<foundry.abstract._Document | null, foundry.data.fields.DataSchema>[],
+        operation: ItemPF2eDatabaseUpdateOperation,
+        user: foundry.documents.BaseUser<foundry.documents.BaseActor<null>>,
+    ): Promise<boolean | void> {
+        if ((await super._preUpdateOperation(documents, operation, user)) === false) {
+            return false;
+        }
+
+        // If this item is of a certain type and belongs to a PC, store current hp to be checked later
+        const actor = documents.find((d): d is ItemPF2e => d instanceof ItemPF2e)?.actor;
+        if (actor) {
+            operation.previousSystemData = fu.mergeObject(operation.previousSystemData ?? {}, {
+                maxHitPoints: actor.hitPoints?.max,
```

Comment:
prepared value?

Reply:
You mean to use `actor.system.attributes.hp.max` instead? Actor is a document, so it should be the prepared value already I think.

Reply:
I mean why are you using the prepared value and not the source value?